pkg/repositories/transformers/task_execution.go

```diff
@@ -151,6 +151,15 @@ func CreateTaskExecutionModel(ctx context.Context, input CreateTaskExecutionMode
 		EventVersion: input.Request.Event.EventVersion,
 	}
 
+	if len(input.Request.Event.Reason) > 0 {
```

Comment:
is this right? should it not check for `len(closure.Reasons) <= 0`

Reply:
so this is run in `CreateTaskExecutionModel` so `closure.Reasons` will always be 0 because we just created the closure. This just says - don't add a reason to the time-series if the reason is an empty string.